components/function-controller/internal/controllers/serverless/function_reconcile.go

```diff
@@ -169,51 +169,73 @@ func (r *FunctionReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error
+		if updateErr := r.updateStatusWithoutRepository(ctx, instance, serverlessv1alpha1.Condition{
 			Type:               serverlessv1alpha1.ConditionConfigurationReady,
 			Status:             corev1.ConditionFalse,
 			LastTransitionTime: metav1.Now(),
 			Reason:             serverlessv1alpha1.ConditionReasonSourceUpdateFailed,
 			Message:            errMsg,
-		})
+		}); updateErr != nil {
+			return ctrl.Result{}, errors.Wrap(updateErr, "while updating status")
+		}
+		return result, err
 	}
 
 	rtmCfg := fnRuntime.GetRuntimeConfig(instance.Spec.Runtime)
 	rtm := fnRuntime.GetRuntime(instance.Spec.Runtime)
+	var result ctrl.Result
 
 	switch {
-	case instance.Spec.Type == serverlessv1alpha1.SourceTypeGit && r.isOnSourceChange(instance, revision):
-		return r.onSourceChange(ctx, instance, &serverlessv1alpha1.Repository{
+	case instance.Spec.Type == serverlessv1alpha1.SourceTypeGit &&
+		r.isOnSourceChange(instance, revision):
+		err = r.onSourceChange(ctx, instance, &serverlessv1alpha1.Repository{
 			Reference: instance.Spec.Reference,
 			BaseDir:   instance.Spec.Repository.BaseDir,
 		}, revision)
-	case instance.Spec.Type != serverlessv1alpha1.SourceTypeGit && r.isOnConfigMapChange(instance, rtm, configMaps.Items, deployments.Items, dockerConfig):
-		return r.onConfigMapChange(ctx, log, instance, rtm, configMaps.Items)
-	case instance.Spec.Type == serverlessv1alpha1.SourceTypeGit && r.isOnJobChange(instance, rtmCfg, jobs.Items, deployments.Items, gitOptions, dockerConfig):
-		return r.onGitJobChange(ctx, log, instance, rtmCfg, jobs.Items, gitOptions, dockerConfig)
-	case instance.Spec.Type != serverlessv1alpha1.SourceTypeGit && r.isOnJobChange(instance, rtmCfg, jobs.Items, deployments.Items, git.Options{}, dockerConfig):
+
+	case instance.Spec.Type != serverlessv1alpha1.SourceTypeGit &&
+		r.isOnConfigMapChange(instance, rtm, configMaps.Items, deployments.Items, dockerConfig):
+		err = r.onConfigMapChange(ctx, log, instance, rtm, configMaps.Items)
+
+	case instance.Spec.Type == serverlessv1alpha1.SourceTypeGit &&
+		r.isOnJobChange(instance, rtmCfg, jobs.Items, deployments.Items, gitOptions, dockerConfig):
+		result, err = r.onGitJobChange(ctx, log, instance, rtmCfg, jobs.Items, gitOptions, dockerConfig)
+
+	case instance.Spec.Type != serverlessv1alpha1.SourceTypeGit &&
+		r.isOnJobChange(instance, rtmCfg, jobs.Items, deployments.Items, git.Options{}, dockerConfig):
 		return r.onJobChange(ctx, log, instance, rtmCfg, configMaps.Items[0].GetName(), jobs.Items, dockerConfig)
+
 	case r.isOnDeploymentChange(instance, rtmCfg, deployments.Items, dockerConfig):
-		return r.onDeploymentChange(ctx, log, instance, rtmCfg, deployments.Items, dockerConfig)
+		result, err = r.onDeploymentChange(ctx, log, instance, rtmCfg, deployments.Items, dockerConfig)
+
 	case r.isOnServiceChange(instance, services.Items):
-		return r.onServiceChange(ctx, log, instance, services.Items)
+		err = r.onServiceChange(ctx, log, instance, services.Items)
```

Comment:
I think that we can return the result immediately.

Reply:
I think it's more readable this way.